googleapiclient/discovery.py

```diff
@@ -774,9 +774,21 @@ def method(self, **kwargs):
     if media_filename:
       # Ensure we end up with a valid MediaUpload object.
       if isinstance(media_filename, six.string_types):
-        (media_mime_type, encoding) = mimetypes.guess_type(media_filename)
+        # FIXME: get this from kwargs['media_mime_type'], rather than mucking with body
+        # we don't currently need encoding, so we don't try to extract it
+        media_mime_type = None
+        if body_value and body_value['metadata']:
+          metadata = body_value['metadata']
+          for key in metadata:
+            if key.lower() == 'content-type':
+              media_mime_type = metadata[key]
+              break
+
+        if media_mime_type is None:
+          (media_mime_type, encoding) = mimetypes.guess_type(media_filename)
```

Comment:
Ok. This line was pre-existing. All I did was indent it, but I can do that stylistic change.